### DISCUSSION TITLE
Auto re-issue

### DISCUSSION THREAD

**Discussion by octavioj** [helgeerbe/certbot-dns-ionos#7] [Ideas]
I see some systems that use Letsencrypt that can re-issue the certificates automatically. Have you thought of creating a version of this that does the same in a docker container that runs continuously? That could also be included in some sort of docker compose feature for some of these systems.

**Comment by helgeerbe** [comment]
Hi, I'm not sure, that I understand you correctly. This is a plugin for certbot for a dns-01 challenge. So certbot itself is the program, which is doing the renewal of the certificate. I run certbot as a systemd service and in its deployment hooks I distribute the certificate do all my containers and other servers.  But it should be quite easy to build a docker container running certbot with this plugin.

**Reply by octavioj** [reply]
Yes, building the container is already in your instructions, which are excellent. But the certificates have an expiration date and for now they need to be manually re-issued, if I understand it correctly. Or is that not the case?

**Comment by helgeerbe** [comment]
certbot is the program whicht is used for the renewal. It checks every day if the certificates are due for renewal. The certificates are valid vor 90 days and can be renewed 30 days before end. certbot is doing the checks and deploys the certificates to your containers if you like.
My certbot plugin is very special for my dns provider ionos.de. Which allows you to retrieve dns wildcard certificates. 
So I think you should start with certbot itself.

**Reply by octavioj** [reply]
I think I understand it now. We also use IONOS as a dynamic DNS provider and I think I understand it now. I just have to make it work. Thank you.
